### PR TITLE
Issue1/basic map

### PR DESCRIPTION
**Pull request by RyanMarshall5765**
Not sure if I should elaborate on what was done so let me know if I should do this. Assuming commit messages should explain it. Also, found out that the App.js is required for create-react-app something to do with node-modules needing it. Have to pass components into the App.js then render through index.js.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,14 +9,24 @@
...
+    "react-leaflet-markercluster": "^2.0.0-rc3",
     "react-scripts": "1.1.5"
   },
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
     "test": "react-scripts test --env=jsdom",
     "eject": "react-scripts eject"
+  },
+  "devDependencies": {
```
Comment (joephayes) [inline package.json]: Missing dependencies for eslint:

```
warning " > eslint-config-airbnb@17.1.0" has unmet peer dependency "eslint@^4.19.1 || ^5.3.0".
warning " > eslint-config-airbnb@17.1.0" has unmet peer dependency "eslint-plugin-import@^2.14.0".
warning " > eslint-config-airbnb@17.1.0" has unmet peer dependency "eslint-plugin-jsx-a11y@^6.1.1".
warning " > eslint-config-airbnb@17.1.0" has unmet peer dependency "eslint-plugin-react@^7.11.0".
warning "eslint-config-airbnb > eslint-config-airbnb-base@13.1.0" has unmet peer dependency "eslint@^4.19.1 || ^5.3.0".
warning "eslint-config-airbnb > eslint-config-airbnb-base@13.1.0" has unmet peer dependency "eslint-plugin-import@^2.14.0".
```

File: src/components/InteractiveMap.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      center: [37.73, 14.2],
+      zoom: 8,
+      maxZoom: 18
+    };
+  }
+  render() {
+    return (
+      <div>
+        <Map center={this.state.center} zoom={this.state.zoom} maxZoom={this.state.maxZoom}>
+          <TileLayer url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png" />
```
Comment (joephayes) [inline src/components/InteractiveMap.js]: Add "attribution" property. See https://github.com/PaulLeCam/react-leaflet/blob/master/example/components/simple.js#L16

File: public/index.html [hunk 3]
```diff
@@ -1,40 +1,20 @@
...
-    <noscript>
-      You need to enable JavaScript to run this app.
-    </noscript>
-    <div id="root"></div>
-    <!--
-      This HTML file is a template.
-      If you open it directly in the browser, you will see an empty page.
+<head>
+  <meta charset="utf-8">
+  <script src="https://raw.github.com/calvinmetcalf/leaflet-ajax/master/dist/leaflet.ajax.min.js"></script>
```
Comment (joephayes) [inline public/index.html]: Replace with CDN - https://cdnjs.com/libraries/leaflet-ajax (it's bad practice to link to GitHub)

File: public/index.html [hunk 4]
```diff
@@ -1,40 +1,20 @@
...
+  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/normalize/7.0.0/normalize.min.css">
```
Comment (joephayes) [inline public/index.html]: Latest version for normalize on cdnjs is 8.0.0 - any reason we cannot use the latest?

File: public/index.html [hunk 5]
```diff
@@ -1,40 +1,20 @@
...
+  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/leaflet/1.3.1/leaflet.css">
```
Comment (joephayes) [inline public/index.html]: update to version 1.3.4 to match package.json

File: src/components/InteractiveMap.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+          <MarkerClusterGroup>
+            <GeoJSON data={geojson} />
```
Comment (joephayes) [inline src/components/InteractiveMap.js]: Can we add a tooltip or very basic popup that displays the name of the location?

File: src/components/InteractiveMap.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+import { Map, TileLayer, GeoJSON } from 'react-leaflet';
+import '../containers/App.css';
+import '../data/geojson';
+import { geojson } from '../data/geojson';
+import MarkerClusterGroup from 'react-leaflet-markercluster';
+
+class InteractiveMap extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
```
Comment (joephayes) [inline src/components/InteractiveMap.js]: It feels like these settings should come from a config file

File: src/components/InteractiveMap.js (hunk 2 above)
Comment (joephayes) [inline src/components/InteractiveMap.js]: We cannot use the Open Street Map tile server - see https://operations.osmfoundation.org/policies/tiles/. We have to chose another tile provider - Mapbox, Carto or Stamen?

Comment (joephayes) [inline src/components/InteractiveMap.js]: I'd also move the tileset URL into a config file

**Comment by joephayes** [comment]
> Not sure if I should elaborate on what was done so let me know if I should do this. Assuming commit messages should explain it.

Always good to add a quick summary in the pull request description.

**Comment by joephayes** [comment]
Note on branch naming. In general let's use this template:

```
[dev initials]-[issue number OR description]
```

so for example:

```
rm-issue1
```
or

```
rm-basic-map
```

**Comment by joephayes** [comment]
BTW - I like the nice, clear commit messages! :-)

File: package.json (hunk 1 above)
Comment (RyanMarshall5765) [reply package.json]: Realized that for some reason they aren't in the commit but are in my local. Would you know a reason for why this would occur?

File: public/index.html (hunk 4 above)
Comment (RyanMarshall5765) [reply public/index.html]: Latest works. Found this in some documentation assuming theirs wasn't updated.

File: package.json (hunk 1 above)
Comment (joephayes) [reply package.json]: Is it possible you added them after you committed the file? Or was the file not saved?

Comment (joephayes) [reply package.json]: Is it possible you added them after you committed the file? Or was the file not saved?

File: src/components/InteractiveMap.js (hunk 2 above)
Comment (RyanMarshall5765) [reply src/components/InteractiveMap.js]: Using Stamen for now. Mapbox updated from Mapbox.js to Mapbox GL JS which is no longer built on leaflet. They deprecated mapbox-studio for developing with leaflet, now only works for the new Mapbox GL JS. Trying to get mapbox-studio-classic working to turn streets off and unneeded icons. Have to build the studio ourselves (https://github.com/mapbox/mapbox-studio-classic) but as of 10 days ago people have been running into issues building it. Sadly, it is no longer supported waiting to see if someone finds a fix in the community.

Comment (RyanMarshall5765) [reply src/components/InteractiveMap.js]: Done

File: package.json (hunk 1 above)
Comment (RyanMarshall5765) [reply package.json]: There were added before. Otherwise, VSCode would allow me to stage it right now. The file and working tree are the same. DevDependencies were not there until I added the ESLint stuff for Airbnb and prettier. For some reason, it committed that object as blank? or so it would seem.

Comment (joephayes) [reply package.json]: Weird...

Comment (RyanMarshall5765) [reply package.json]: Added some of the things it said it was missing. The missing ones from the file are now there and not showing as an add. I am getting an error myself though. I can't add ESLint to the dependencies because its part of create-react-app so it throws an error. Probably don't want to eject as well.

**Comment by RyanMarshall5765** [comment]
ESLint wouldn't add it was an issue on my end. Node wasn't updated in a long time so I couldn't use the newer versions. Thus could not add it. Should be all fixed now.

File: src/components/InteractiveMap.js (hunk 7 above)
Comment (joephayes) [reply src/components/InteractiveMap.js]: I'll open a new issue for this.

File: src/components/InteractiveMap.js (hunk 2 above)
Comment (joephayes) [reply src/components/InteractiveMap.js]: I'll add a new issue for this

**Comment by joephayes** [comment]
Please "Squash and Merge" using the Github button

screen shot 2018-09-05 at 8 31 45 am
